torch/csrc/jit/runtime/symbolic_script.cpp

```diff
@@ -1014,63 +1014,19 @@ const std::vector<std::string> functions = {
             return output, backward
 
         # disable the layernorm AD temporarily because of bug in https://github.com/pytorch/pytorch/issues/19769
-        def layer_norm_disabled(input : Tensor,
+        def layer_norm(input : Tensor,
                        normalized_shape : List[int],
                        weight : Optional[Tensor],
                        bias : Optional[Tensor],
                        eps : float,
                        cudnn_enable : bool):
 
-            input_ndim = input.dim()
-            normalized_ndim = len(normalized_shape)
-            n = 1
-            for i in range(input_ndim - normalized_ndim):
-                n *= input.size(i)
-
-            input_reshape = input.contiguous().view(1, n, -1)
-
-            bn_out, save1, save2, reserve, impl_idx = torch._batch_norm_impl_index(
-                input_reshape, None, None, None, None, True,
-                0.0, eps, cudnn_enable)
-
-            bn_out = bn_out.view(input.size())
-            if weight is not None and bias is not None:
-                output = bias.addcmul(bn_out, weight, value=1)
-            elif weight is not None:
-                output = bn_out.mul(weight)
-            elif bias is not None:
-                output = bn_out.add(bias)
-            else:
-                output = bn_out
-
-            def backward(grad_output):
-                if weight is not None and bias is not None:
-                    grad_bn_out = grad_output * weight
-                    grad_weight = (grad_output * bn_out)._grad_sum_to_size(weight.size())
-                    grad_bias = grad_output._grad_sum_to_size(bias.size())
-                elif weight is not None:
-                    grad_bn_out = grad_output * weight
-                    grad_weight = (grad_output * bn_out)._grad_sum_to_size(weight.size())
-                    grad_bias = None
-                elif bias is not None:
-                    grad_bn_out = grad_output
-                    grad_weight= None
-                    grad_bias = grad_output._grad_sum_to_size(bias.size())
-                else:
-                    grad_bn_out = grad_output
-                    grad_weight= None
-                    grad_bias = None
-
-
-                grad_bn_out = grad_bn_out.contiguous().view(1, n, -1)
-
-                grad_input, _, _ = torch._batch_norm_impl_index_backward(
-                    impl_idx, input_reshape, grad_bn_out, None, None, None,
-                    save1, save2, True, eps, [True, False, False], reserve)
-
-                grad_input = grad_input.view(input.size())
+            output, mean, rstd = torch.native_layer_norm(input, normalized_shape, weight, bias, eps)
+            def backward(grad_output):
+                # TODO: grad_bias and grad_weight are disabled because we are missing multiple kernel support
```

Comment:
Even though we can't do multiple kernel yet, at least let's copy over the aten implementation so we are not screwing it when we upstream this:
```
                if weight is not None:	
                    grad_bn_out = grad_output * weight	
                    grad_weight = (grad_output * bn_out)._grad_sum_to_size(weight.size())	
                    grad_bias = None	
                elif bias is not None:	
                    grad_bn_out = grad_output	
                    grad_weight= None	
                    grad_bias = grad_output._grad_sum_to_size(bias.size())
```

Reply:
bn_out might be tricky & not performing. But let's hope that we don't have affine in bert :laughing:

Reply:
```
if weight is not None:	
  grad_weight = (grad_output * output)._grad_sum_to_size(weight.size())	

if bias is not None:	
  grad_bias = grad_output._grad_sum_to_size(bias.size())
```

`bn_out` and `grad_output` should have the same shape.